chandra_aca/planets.py

```diff
@@ -586,3 +591,132 @@ def get_planet_chandra_horizons(
     del dat["null3"]
 
     return dat
+
+
+@dataclass
+class EarthBoresightAngles:
+    """Earth boresight angles from Chandra
+
+    Attributes
+    ----------
+    times : CxoTime
+        Times
+    earth_limb_angle : ndarray
+        Earth limb angle (deg)
+    earth_open_angle : ndarray
+        Earth opening angle (deg)
+    earth_center_angle : ndarray
+        Earth center angle (deg)
+    """
+
+    times: CxoTime
+    earth_limb_angle: np.ndarray
+    earth_open_angle: np.ndarray
+    earth_center_angle: np.ndarray
+
+
+def get_earth_boresight_angle(start: CxoTimeLike, stop: CxoTimeLike):
+    """Calculate Earth boresight angle from Chandra.
+
+    The key calculations were adapted from acis_taco.calc_earth_vis() so there is some
+    heritage.
+
+    Parameters
+    ----------
+    start : CxoTime-compatible
+        Start time
+    stop : CxoTime-compatible
+        Stop time
+
+    Returns
+    -------
+    EarthBoresightAngles
+        Dataclass with attributes:
+
+        - times (CxoTime)
+        - earth_limb_angle (deg)
+        - earth_open_angle (deg)
+        - earth_center_angle (deg)
+    """
+    from cheta import fetch
+
+    RAD_EARTH = 6371000.0  # m
+    start = CxoTime(start)
+    stop = CxoTime(stop)
+
+    msids_ephem = fetch.MSIDset(
+        ["orbitephem0_x", "orbitephem0_y", "orbitephem0_z"],
+        start - 15 * u.min,
+        stop + 15 * u.min,
+    )
+
+    msid_q_att = fetch.Msid("quat_aoattqt", start, stop)
+
+    q_att = msid_q_att.vals
+    q_att_times = msid_q_att.times
+
+    ephem_times = msids_ephem["orbitephem0_x"].times
+    p_chandra_eci = np.array(
+        [
+            np.interp(
+                x=q_att_times,
+                xp=ephem_times,
+                fp=msids_ephem[f"orbitephem0_{axis}"].vals,
+            )
+            for axis in ["x", "y", "z"]
+        ]
+    ).transpose()
+
+    q_att_inv = q_att.transform.swapaxes(1, 2)
+    p_earth_body = np.einsum("ijk,ik->ij", q_att_inv, -p_chandra_eci)
```

Comment:
Just noting that the previous two lines are equivalent to the following line (swapping the indices within `np.einsum`):
```python
p_earth_body = np.einsum("ikj,ik->ij", q_att.transform, -p_chandra_eci)
```

(I'm not sure the two lines make it more understandable)